index/secondary-index/src/test/scala/org/apache/carbondata/spark/testsuite/secondaryindex/TestSIWithSecondaryIndex.scala

```diff
@@ -434,6 +434,38 @@ class TestSIWithSecondaryIndex extends QueryTest with BeforeAndAfterAll {
     assert(df.exists(_.get(0).toString.contains("nam(e")))
   }
 
+  test("test alter table set streaming for table having SI") {
+    sql("drop table if exists maintable2")
+    sql("create table maintable2 (a string,b string,c int) STORED AS carbondata ")
+    sql("insert into maintable2 values('k','x',2)")
+    sql("create index m_indextable on table maintable2(b) AS 'carbondata'")
+    var exeption = intercept[RuntimeException] {
```

Comment:
IMO Intercepting specific exception - `UnsupportedOperationException` would be good rather than generic `RuntimeException`.

Reply:
In `AlterTableUtil` line 541 we are catching the exception and making it sys.error(). Because of this it is throwing it as `RunTimeException`.